### PR TITLE
Do not scope AMQP adapter client to single tenant

### PR DESCRIPTION
**Pull request by sophokles73**
The AMQP adapter will authorize the client's requests to open links and
send messages based on the credentials that the client has used to
establish the connection. Thus, there is no need to scope the client
itself to a tenant.

Also removed the need for getting/creating a dedicated sender object in favor of having dedicated methods for sending messages.
The interface variants that allow passing in a SpanContext have been removed in favor of simply adding an optional SpanContext typed parameter to the message sending methods.

### CODE REVIEW

**Comment by sophokles73** [comment]
@calohmn would you mind taking a look?

File: clients/device-amqp/src/main/java/org/eclipse/hono/client/device/amqp/impl/AmqpAdapterClientCommandConsumer.java [hunk 1]
```diff
@@ -71,12 +71,11 @@ public static Future<CommandConsumer> create(
             final BiConsumer<ProtonDelivery, Message> messageHandler) {
 
         Objects.requireNonNull(con);
-        Objects.requireNonNull(tenantId);
```
Comment (calohmn) [inline clients/device-amqp/src/main/java/org/eclipse/hono/client/device/amqp/impl/AmqpAdapterClientCommandConsumer.java:74]: Javadoc should be adapted here as well.

File: clients/device-amqp/src/main/java/org/eclipse/hono/client/device/amqp/AmqpAdapterClient.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+     * @param tenantId The tenant that the device belongs to, or {@code null} to determine the tenant from 
+     *                 the device that has authenticated to the AMQP adapter.
+     *                 Unauthenticated clients must provide a non-{@code null} value to indicate the tenant of the
+     *                 device to consume commands for.
+     * @param deviceId The identifier of the device to consume commands for, or {@code null} to consume commands
+     *                 for the device that has authenticated to the AMQP adapter.
+     *                 Authenticated gateway devices can use this parameter to consume commands for another device
+     *                 that the gateway is authorized to act on behalf of.
+     *                 Unauthenticated clients must provide a non-{@code null} value to indicate the device to
+     *                 consume commands for.
```
[the remark this answers is not in the document]
Comment (sophokles73) [reply clients/device-amqp/src/main/java/org/eclipse/hono/client/device/amqp/AmqpAdapterClient.java:73]: Yes, I guess you are right.

**Review by calohmn — APPROVED** [review]
LGTM

**Pull request merged**